test/transforms.jl

```diff
@@ -203,18 +211,48 @@
     # reject columns whose names contain the character x
     T = Reject(r"x")
     n, c = apply(T, t)
-    @test Tables.columnnames(n) == (:y1, :y2)
+    @test Tables.columnnames(n) == [:y1, :y2]
     tₒ = revert(T, n, c)
     @test t == tₒ
 
     # reject columns whose names contain the character y
     T = Reject(r"y")
     n, c = apply(T, t)
-    @test Tables.columnnames(n) == (:x1, :x2)
+    @test Tables.columnnames(n) == [:x1, :x2]
     tₒ = revert(T, n, c)
     @test t == tₒ
   end
 
+  @testset "TableSelection" begin
+    a = rand(4000)
+    b = rand(4000)
+    c = rand(4000)
+    d = rand(4000)
+    e = rand(4000)
+    f = rand(4000)
+    t = Table(; a, b, c, d, e, f)
+
+    # Tables.jl interface
+    s = TableTransforms.TableSelection(t, [:a, :b, :e])
+    @test Tables.istable(s) == true
+    @test Tables.columnaccess(s) == true
+    @test Tables.rowaccess(s) == false
+    @test Tables.columns(s) === s
+    @test Tables.columnnames(s) == [:a, :b, :e]
+    @test Tables.schema(s).names == (:a, :b, :e)
+    @test Tables.schema(s).types == (Float64, Float64, Float64)
+    @test Tables.materializer(s) == Tables.materializer(t)
+
+    # getcolumn
+    cols = Tables.columns(t)
+    @test Tables.getcolumn(s, :a) == Tables.getcolumn(cols, :a)
+    @test Tables.getcolumn(s, 1) == Tables.getcolumn(cols, 1)
+    @test Tables.getcolumn(s, 3) == Tables.getcolumn(cols, :e)
+
+    # throws
+    @test_throws AssertionError TableTransforms.TableSelection(t, [:a, :b, :z])
+  end
```

Comment:
Amazing software engineering skills. That really helps in the long term.

Reply:
Thanks!